wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 1 - Refactor use of Repository in favor of Prefs for Selected Site

Parent #15191

This PR mainly refactors all usages of `AppPrefs.get/setSelectedSite(...)` to go through the `SelectedSiteRepository` and through that change to make this repository the source of truth for when a selected site is being accessed.

Also, as part of this PR, the below occurred:
- Documentation was added to `AppPrefs.get/setSelectedSite(...)` method calls so that access to a selected site becomes explicitly prohibited and only available through the `SelectedSiteRepository`.
- All `id` naming conventions, mainly the local id ones, for when a selected site is accessed, were renamed to `siteLocalId`.
- Two new FAQ questions was added based on both those main changes:
  - Should I use `AppPrefs` or `SelectedSiteRepository` as the source of truth on a selected site?
  - Should I use `id`, `siteId`, `siteLocalId`, `siteRemoteId` or any other naming convention?

-----

To test:
- Smoke test the `Login` screen. Focus on site selection, site creation and the quick start functionality.
- Smoke test the `My Site` screen. Focus on site selection, site creation and the quick start functionality.
- Smoke test the `Stats` screen. Just verify that everything works as expected and there are no surprises.
- Smoke the the `Reader` screen. Just verify that everything works as expected and there are no surprises.
- Smoke test the `Media Upload` functionality. Just verify that everything works as expected and there are no surprises.

-----

Merge Instructions for @ashiagr :
- ~~Wait for `18.1` code freeze, cut and for the `release/18.1` branch to happen.~~ ✅ 
- ~~Remove the `[Status] Not Ready for Merge` label.~~ ✅ 
- ~~Remove the `(DO NOT MERGE)` suffix from the title.~~ ✅ 
- ~~Merge this PR.~~ ✅ 

-----

## Regression Notes
1. Potential unintended areas of impact

The `Login`, `My Site`, `Stats`, `Reader` and/or `Media Upload` logic might be somehow affected by all the changes in this PR since there has been some updates to all that.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
